From 3d59dfe418696d24b43242d52c4ec49aeb1e3c48 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v2-14.17-3d59dfe4186-20230328T110947Z-avarab@gmail.com>
In-Reply-To: <cover-v2-00.17-00000000000-20230328T110946Z-avarab@gmail.com>
References: <cover-00.17-00000000000-20230317T152724Z-avarab@gmail.com>
	<cover-v2-00.17-00000000000-20230328T110946Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 16 Mar 2023 16:22:21 +0100
Subject: [PATCH v2 14/17] cocci: apply the "rerere.h" part of
 "the_repository.pending"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Apply the part of "the_repository.pending.cocci" pertaining to
"rerere.h".

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 builtin/stash.c                                 | 2 +-
 contrib/coccinelle/the_repository.cocci         | 4 ++++
 contrib/coccinelle/the_repository.pending.cocci | 4 ----
 rerere.h                                        | 3 ---
 4 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/builtin/stash.c b/builtin/stash.c
index 60052ec9d7b..b60c82f7af2 100644
--- a/builtin/stash.c
+++ b/builtin/stash.c
@@ -602,7 +602,7 @@ static int do_apply_stash(const char *prefix, struct stash_info *info,
 		ret = error(_("could not write index"));
 
 	if (ret) {
-		rerere(0);
+		repo_rerere(the_repository, 0);
 
 		if (index)
 			fprintf_ln(stderr, _("Index was not unstashed."));
diff --git a/contrib/coccinelle/the_repository.cocci b/contrib/coccinelle/the_repository.cocci
index b20a3e70845..1d1ac7d4fc5 100644
--- a/contrib/coccinelle/the_repository.cocci
+++ b/contrib/coccinelle/the_repository.cocci
@@ -109,6 +109,10 @@
 |
 - dwim_ref
 + repo_dwim_ref
+// rerere.h
+|
+- rerere
++ repo_rerere
 )
   (
 + the_repository,
diff --git a/contrib/coccinelle/the_repository.pending.cocci b/contrib/coccinelle/the_repository.pending.cocci
index 8bf30798a04..1190a3312bd 100644
--- a/contrib/coccinelle/the_repository.pending.cocci
+++ b/contrib/coccinelle/the_repository.pending.cocci
@@ -5,11 +5,7 @@
 @@
 @@
 (
-// rerere.h
-- rerere
-+ repo_rerere
 // revision.h
-|
 - init_revisions
 + repo_init_revisions
 )
diff --git a/rerere.h b/rerere.h
index c32d79c3bd8..876c316321e 100644
--- a/rerere.h
+++ b/rerere.h
@@ -24,9 +24,6 @@ struct rerere_id {
 };
 
 int setup_rerere(struct repository *,struct string_list *, int);
-#ifndef NO_THE_REPOSITORY_COMPATIBILITY_MACROS
-#define rerere(flags) repo_rerere(the_repository, flags)
-#endif
 int repo_rerere(struct repository *, int);
 /*
  * Given the conflict ID and the name of a "file" used for replaying
-- 
2.40.0.rc1.1034.g5867a1b10c5

